### ISSUE TITLE
Attempting to Upgrade Chart to Enable Examples Causes Postgres Error

### ISSUE THREAD

**Issue by victor-perchwell** [apache/airflow#29444]
### Official Helm Chart version

1.8.0 (latest released)

### Apache Airflow version

2.5.1

### Kubernetes Version

v1.26.1

### Helm Chart configuration

I'm using the official helm chart and modifying the values.yaml file to enable examples by uncommenting:
![image](https://user-images.githubusercontent.com/105374838/217901076-bf0ff09b-aa14-48b9-ad8c-85ee70ea953e.png)

### What happened

When I attempt to upgrade the release to use the modified values.yaml file, I get the following error:
![image](https://user-images.githubusercontent.com/105374838/217901295-6317e3fa-8726-48b9-80a4-8ef0f3ec87db.png)

I'm extracting the values.yaml to modify via:
```
helm-show-values apache-airflow/airflow > values.yaml
```

### How to reproduce

Deploy the official Airflow Helm chart to minikube, attempt to modify values.yaml to turn examples on, and upgrade the release with the new values.yaml file.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by iuriqc** [comment]
I had a similar situation. When modifying the yaml to use gitsync and updating Airflow I had the same error. I tried modifying the "airflow-postgresql" secret to include the "password" key and the helm even completes the update, but postgres doesn't work properly. Following [this orientation](https://github.com/apache/airflow/blob/main/chart/RELEASE_NOTES.rst#airflow-helm-chart-1-8-0-2023-02-06) to delete the secret and statefulset also does not work.

**Comment by hussein-awala** [comment]
Is your current version deployed with `1.8.0` or earlier?
If it's an earlier version, you need to update the values file, because we upgraded the postgresql chart from `10.5.3` to `12.1.9` and the schema of the values file ([PR](https://github.com/apache/airflow/pull/29071))

**Comment by empire-sports** [comment]
Commenting on this.  I am upgrading from 1.7 to 1.8.  I used a custom values.yaml file with my implementation to enable the git-sync.  I'm receiving a similar error.

`namespace/colosseum configured
secret/airflow-gke-git-secret configured
history.go:56: [debug] getting history for release airflow
upgrade.go:144: [debug] preparing upgrade for airflow
Error: UPGRADE FAILED: execution error at (airflow/charts/postgresql/templates/secrets.yaml:17:24): 
PASSWORDS ERROR: The secret "airflow-postgresql" does not contain the key "postgres-password"

helm.go:84: [debug] execution error at (airflow/charts/postgresql/templates/secrets.yaml:17:24): 
PASSWORDS ERROR: The secret "airflow-postgresql" does not contain the key "postgres-password"`

**Comment by potiuk** [comment]
Yes. This is a known "expectation" for chart 1.8.0 and the way how to deal with it is described in [release notes](https://airflow.apache.org/docs/helm-chart/stable/release_notes.html#airflow-helm-chart-1-8-0-2023-02-06)

**Comment by agentdanger** [comment]
I can recreate this issue by doing a fresh install using the 1.8 helm image "my-airflow", in a fresh namespace on kubernetes, and customizing the values.yaml to attempt to turn on git-sync.  I have the same issue the original OP is mentioning.

**Comment by agentdanger** [comment]
I get the issue when I do a helm upgrade to install the custom values.

**Comment by agentdanger** [comment]
I was able to resolve this issue.  

Upgrading also refers to upgrading the existing 1.8 instance.  You have to follow @potiuk suggestion within the release notes.

`kubectl delete secret {RELEASE_NAME}-postgresql`
`kubectl delete statefulset {RELEASE_NAME}-postgresql`

Then in your values.yaml file.  You must set the username and password in the postgresql subchart configuration:

postgresql:
  enabled: true
  image:
    tag: "11"
  auth:
    enablePostgresUser: true
    postgresPassword: postgres
    username: **YOUR_USERNAME**
    password: **YOUR_PASSWORD**

No issues once I did all of those steps.

**Comment by agentdanger** [comment]
I'm caveating my "no issue" suggestion above.  I'm currently working through bugs on the git-sync but those could be my own doing.  My work around was able to solve the postgresql error I was getting...

**Comment by agentdanger** [comment]
I was able to solve this issue by installing 1.8 with the configured yaml file.  

My namespace for this implementation was colosseum.

I had a custom yaml filed called values.yaml in the same working directory that I was performing my kubectl commands.
I also had my git secret ready in the same working directory.

kubectl commands I used are below:

`kubectl create namespace colosseum --dry-run=client -o yaml | kubectl apply -f -`
`kubectl create secret generic airflow-gke-git-secret --from-file=gitSshKey=airflow-gke-git-secret --namespace colosseum --dry-run=client -o yaml | kubectl apply -f -`
`helm upgrade --install my-airflow apache-airflow/airflow --namespace colosseum --create-namespace -f values.yaml --debug`
